### ISSUE TITLE
关于antifungal dataset数据缺失

### ISSUE THREAD

**Issue by Li-emo** [jimmyrate/deepAMP#9]
您好，在您提供的脚本里提到了dataset=antifungal，但data目录下没有收录该数据，我想向您请教一下antifungal dataset的数据类型和格式，因为我想优化一个抗真菌的AMP。如果我不想针对真菌优化AMP，在sample_conf中是不是可以删掉dataset=antifungal这个部分。
十分期待您的回复。

**Comment by jimmyrate** [comment]
antifungal dataset是我们另一个未完成的项目，不好意思，这部分数据暂时还不能公开。
如果你不想针对抗真菌肽优化，当然可以在sample_conf中删除这个部分，只需要替换成你想要优化的数据集就行了

**Comment by Li-emo** [comment]
您好，我还有一些问题想向您请教。希望您能指点一下。
1.如果我想利用deepAMP优化我的抗菌肽，是不是需要先运行dataPre.py这个脚本，要先用'uniport_all_60'或'random_pair'数据集finetune然后再用random_chem_penetratin_pair refinetune。那么对应的sample_conf的dataset=就是'uniport_all_60'or'random_pair'数据集，或者random_chem_penetratin_pair数据集吗？那我是不是要把这个脚本运行两次？
2.当我运行到text_path[f'inter_test_{reinforce_conf.optimize_amp}_all']=&nbsp; f'{train_conf.basicPath}reinforcement_proteins_package/{reinforce_conf.optimize_amp}_num_10.txt'
这一步时，报这样的错误FileNotFoundError: [Errno 2] No such file or directory: './output/finetune/dataset=uniport_all_60,seed=2,optimizer=Ranger,batch_size=256,epochs=350,lr=0.0007,weight_decay=0.0001,lr_decay=True,lr_scheduler=WarmupLR,warmupEpochs=30_epoch_349/daset=None,seed=6,optimizer=Ranger,batch_size=128,epochs=200,lr=6e-05,weight_decay=0.001,lr_decay=True,lr_scheduler=WarmupLR,warmupEpochs=15/reinforcement_proteins_package/RQIKIWFQNRRMKWKK_num_10.txt'
是不是因为我的deepAMP-general_seed2_epoch349.pkl文件放错位置导致强化学习过程尚未进行。想请教您，训练模型是否用这个文件，我在deepAMP根目录下创建了output/finetune/路径，我把该训练模型放在finetune文件夹下是否可以。因为实在找不到解决办法，但是真的很需要deepAMP所以来打扰您。
十分期待您的回复。
祝好
&nbsp;


李艺宣
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年5月10日(星期六) 中午1:28
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [jimmyrate/deepAMP] 关于antifungal dataset数据缺失 (Issue #9)












jimmyrate left a comment (jimmyrate/deepAMP#9)
 
antifungal dataset是我们另一个未完成的项目，不好意思，这部分数据暂时还不能公开。
 如果你不想针对抗真菌肽优化，当然可以在sample_conf中删除这个部分，只需要替换成你想要优化的数据集就行了

**Comment by jimmyrate** [comment]
1.不需要运行dataPre.py，由于我这个代码是2021年左右写的，可能写的有点烂。是先在dataPre.py确定你要加载哪个模型的cpkt，这个加载哪个模型的ckpt是由dataset=‘**’决定的。
然后你在sample_conf中设置好，你待优化序列的路径，最后运行do_sample.py文件来进行优化。

2.关于这个问题，你想直接使用模型优化的话，我建议您运行do_sample.py，然后debug看看是不是你哪部分路径没有设置对。因为这个代码写的确实有点难懂，你可以这样理解，dataPre.py中是设置路径和读取预训练或者微调模型的ckpt的。但是你要优化自己的候选集，你先需要设置好读取模型的路径，和你待优化的文件路径。然后运行do_sample.py进行优化。

如果您还有问题，可以添加我的wechat:109161772

**Comment by Li-emo** [comment]
天呐，太感谢您了，我想加一下您的微信，谢谢



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2025年5月10日(周六) 下午4:00
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;;"State ***@***.***&gt;;
主题: Re: [jimmyrate/deepAMP] 关于antifungal dataset数据缺失 (Issue #9)


 jimmyrate left a comment (jimmyrate/deepAMP#9)
 
1.不需要运行dataPre.py，由于我这个代码是2021年左右写的，可能写的有点烂。是先在dataPre.py确定你要加载哪个模型的cpkt，这个加载哪个模型的ckpt是由dataset=‘**’决定的。
 然后你在sample_conf中设置好，你待优化序列的路径，最后运行do_sample.py文件来进行优化。
 
2.关于这个问题，你想直接使用模型优化的话，我建议您运行do_sample.py，然后debug看看是不是你哪部分路径没有设置对。因为这个代码写的确实有点难懂，你可以这样理解，dataPre.py中是设置路径和读取预训练或者微调模型的ckpt的。但是你要优化自己的候选集，你先需要设置好读取模型的路径，和你待优化的文件路径。然后运行do_sample.py进行优化。
 
如果您还有问题，可以添加我的wechat:109161772

**Comment by Li-emo** [comment]
不好意思，我刚刚尝试添加109161772这个微信号没有找到该用户，可否麻烦您再发一次微信账号呢。我是中科院遗传所的一名博士生。
&nbsp;


李艺宣
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年5月10日(星期六) 下午4:00
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [jimmyrate/deepAMP] 关于antifungal dataset数据缺失 (Issue #9)












jimmyrate left a comment (jimmyrate/deepAMP#9)
 
1.不需要运行dataPre.py，由于我这个代码是2021年左右写的，可能写的有点烂。是先在dataPre.py确定你要加载哪个模型的cpkt，这个加载哪个模型的ckpt是由dataset=‘**’决定的。
 然后你在sample_conf中设置好，你待优化序列的路径，最后运行do_sample.py文件来进行优化。
 
2.关于这个问题，你想直接使用模型优化的话，我建议您运行do_sample.py，然后debug看看是不是你哪部分路径没有设置对。因为这个代码写的确实有点难懂，你可以这样理解，dataPre.py中是设置路径和读取预训练或者微调模型的ckpt的。但是你要优化自己的候选集，你先需要设置好读取模型的路径，和你待优化的文件路径。然后运行do_sample.py进行优化。
 
如果您还有问题，可以添加我的wechat:109161772

**Comment by jimmyrate** [comment]
不好意思，之前应该是权限关闭了，麻烦您再次添加试试
